### DISCUSSION TITLE
fetch data from MongoDB Compass and save to state

### DISCUSSION THREAD

**Discussion by maksymvasylyev** [neuefische/questions#621] [Web] [answered]
Hello
I fetched data from Mongo DataBase and saved it to state. Everything works well, but by page reloading all changes in my App are deleted.
That happens, because starting point of my state is []. And I don`t know how to prevent it.
I want to fetch data only once and than work only with my state and local storage.

I am including PR:

https://github.com/maksymvasylyev/capstone-project/pull/30

Thank you for your help.

Max

**Comment by mbosselmann** [comment, marked as answer]
We discussed the solution together (working with useEffect and a fetch function) and also discussed if it is necessary to work with a database at this point and how the code should be refactored if so.

**Reply by maksymvasylyev** [reply]
Thank you very much for helping in resolving my problem 👍🏼
